File: Raven.Studio.Html5/App/models/filesystem/synchronizationReport.ts
class synchronizationReport implements documentBase {

    FileName: string;
    fileETag: string;
    Type: filesystemSynchronizationType;

    bytesTransfered: number;
    bytesCopied: number;
    needListLength: number;

    exception: any;

    constructor(dto: filesystemSynchronizationReportDto) {
        this.FileName = dto.FileName;
        this.fileETag = dto.FileETag;
        this.Type = dto.Type;

        this.bytesTransfered = dto.BytesTransfered;
        this.bytesCopied = dto.BytesCopied;
        this.needListLength = dto.NeedListLength;

        this.exception = dto.Exception;
    }

    getId() {
        return this.FileName;
    }

    getDocumentPropertyNames(): Array<string> {
        return ["Id", "Type"];
    }
}

export = synchronizationReport;

File: Raven.Abstractions/FileSystem/SynchronizationReport.cs
using System;
using Raven.Abstractions.Data;
using Raven.Imports.Newtonsoft.Json;

namespace Raven.Abstractions.FileSystem
{
	public class SynchronizationReport
	{
		public SynchronizationReport()
		{
		}

		[JsonConstructor]
		public SynchronizationReport(string fileName, Guid fileETag, SynchronizationType type)
		{
			FileName = fileName;
			FileETag = fileETag;
			Type = type;
		}

		public string FileName { get; private set; }
		public Etag FileETag { get; private set; }
		public long BytesTransfered { get; set; }
		public long BytesCopied { get; set; }
		public long NeedListLength { get; set; }
		public Exception Exception { get; set; }
		public SynchronizationType Type { get; private set; }
	}
}